docs/Course Wiki/CS Course Offerings/CS225.md

```diff
@@ -0,0 +1,90 @@
+# CS225
+
+CS225 (Data Structures) is a 4-credit hour course that is specifically required for CS majors and CEs and qualifies as a 3 of 5 technical elective for EEs. It is offered in both the fall and spring. 
+
+## Content Covered
+
+This course teaches a variety of ways to store collections of data in a computer program and discusses the advantages and disadvantages associated with the different methods. You will learn how to build various data-storage structures, and you’ll discover why you might prefer one over another in a particular situation. The combined arts of design, analysis and justification are the substance of the class. The class assumes basic skills programming in C++. The course will not review the language, but the first MP will mostly be a refresher.
+
+### Topics
+
+- Math Concepts and Basic Boolean Logic
+- Big-O
+- Data Structures
+- Trees and Linked Lists
+- Search Algorithms
+- Sorting Algorithms
+- Image Manipulation
+
+## Prerequisites
+
+- [CS126](./CS126.md) OR [CS128](./CS128.md) OR [ECE220](../ECE%20Course%20Offerings/ECE220.md)
+- [CS173](./CS173.md) OR [CS413](./CS413) OR [MATH213](../MATH%20Course%20Offerings/MATH413.md) OR [MATH347](../MATH%20Course%20Offerings/MATH347.md) OR [MATH412](../MATH%20Course%20Offerings/MATH412.md) OR [MATH413](../MATH%20Course%20Offerings/MATH413.md)
+
+CS225 has two main prerequisites, one being a programming course and one being a discrete structures theory course.  CS128 is typically taken by CS students and ECE 220 is typically taken by ECE students to satisfy the prerequisites.  Either CS173 or MATH213 is recommended to satisfy the discrete math prereq. Both programming and theory are used fairly heavily in the class, so both should be taken beforehand.
+
+## When to Take it
+
+For EE students using this class as one of the 3 out of 5 courses, it doesn't really matter when CS 225 is taken. However, for CompE and CS students, this class is much more important as it is the gateway to many upper level ECE and CS courses. CS 225 should be taken soon after taking ECE 220, and as mentioned earlier, taken after Discrete Math. Generally, most EE and CompE students will take this class during their second semester of their sophomore year, but many also take it during the first semester. For the latter, taking MATH 213 is easier as it does not require a CS 125 or ECE 220 prerequisite unlike CS 173. With the emphasis placed on using software for a variety of work in ECE, this is a fundamental course and will benefit EEs seeking to strengthen their programming skills. Another reason to take this class early is so the student will have an advantage if applying for internships related to software development.
```

Comment:
For the latter, taking MATH 213 is easier as it does not require a CS 125 or ECE 220 prerequisite unlike CS 173.

to

To take CS225 first semester sophomore year, it is easier to use MATH213 as a prerequisite since MATH213 doesn't have a CS124 prerequisite, unlike CS173.

Reply:
maybe mention interviews specifically during advantage section